source/fuzz/transformation_duplicate_region_with_selection.cpp

```diff
@@ -0,0 +1,566 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_duplicate_region_with_selection.h"
+
+#include "source/fuzz/fuzzer_util.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationDuplicateRegionWithSelection::
+    TransformationDuplicateRegionWithSelection(
+        const spvtools::fuzz::protobufs::
+            TransformationDuplicateRegionWithSelection& message)
+    : message_(message) {}
+
+TransformationDuplicateRegionWithSelection::
+    TransformationDuplicateRegionWithSelection(
+        uint32_t new_entry_fresh_id, uint32_t condition_id,
+        uint32_t merge_label_fresh_id, uint32_t entry_block_id,
+        uint32_t exit_block_id,
+        std::map<uint32_t, uint32_t> original_label_to_duplicate_label,
+        std::map<uint32_t, uint32_t> original_id_to_duplicate_id,
+        std::map<uint32_t, uint32_t> original_id_to_phi_id) {
```

Comment:
These maps are passed by value.  You should instead pass them by const reference.

In CLion, clang-tidy gives me a warning about this:

```
Clang-Tidy: The parameter 'original_label_to_duplicate_label' is copied for each invocation but only used as a const reference; consider making it a const reference
```